### ISSUE TITLE
Destructuring is broken in generators.

### ISSUE THREAD

**Issue by z3tsu** [6to5/6to5#617]
**case 1**:
```javascript
function* foo() {
  let [item1, item2] = yield fetchItemsAsync();
}
```
'regenerator' transform throws:
```
{type: ArrayPattern, start: 1673, end: 1684, loc: [object Object], range: 1673,1684, elements: [object Object]} does not match type Expression
```

**case 2**:
```javascript
function* foo() { 
  let items = yield fetchItemsAsync();
  let [item1, item2] = items;
}
```
destructuring stays in generated code and fails in runtime.

**case 3** (object destructuring):
```javascript
function* foo() { 
  let items = yield fetchItemsAsObjectAsync();
  let {item1, item2} = items;
}
```
destructuring is transformed to:
```javascript
var _ref = items;
item1 = _ref.item1;
item2 = _ref.item2;
```
but gets hoisted as:
```javascript
var items;
var { item1: item1, item2: item2 };
```

**Comment by bmeck** [comment]
this seems to be largely tied to order of operations in the transforms, I tried moving them around but regenerator is doing special things with default arguments and rest params that caused the tests to fail.

**Comment by z3tsu** [comment]
@bmeck yep, sebmck tried to hotfix it moving 'es6.destructuring' up, but this change broke other tests.

**Comment by sebmck** [comment]
I'll have another look tonight, it'll require some fiddling around with transformers. Regenerator only supports ES5 constructs which sucks **hard**.

**Comment by sebmck** [comment]
Fixed as of 3.3.0, thanks!

**Comment by assaf** [comment]
Awesome, thanks!
